apps/backend/src/services/configuration_monitoring.py

```
@@ -465,78 +631,87 @@ def _determine_config_type(self, filepath: str) -> str:
+
                 self.logger.debug(f"Stored configuration snapshot: {device_id}:{file_path}")
-                
+
         except Exception as e:
             self.logger.error(f"Failed to store configuration snapshot: {e}")
-            raise DatabaseOperationError(f"Failed to store configuration snapshot: {e}")
+            raise DatabaseOperationError(
+                message="Failed to store configuration snapshot",
+                operation="store_configuration_snapshot",
+                details={"exception": str(e), "file_path": file_path, "config_type": config_type},
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)